### PR TITLE
[C#] Singleton Design Pattern revamp

### PR DESCRIPTION
**Pull request by YoussefWaelMohamedLotfy**
As per the proposal in #18, this is the first implementation for trying out the suggested changes with the community.

## Changes implemented

- Created a dedicated `Singleton` project, with an example in `Program.cs` to show demo code.
- sealed `Logger` class as per best practice for Singleton Design Pattern
- Use File-Scoped namespaces, resulting in one less indentation level
- Respected C# naming conventions, e.g. changing `instance` to `s_instance` as per [IDE1006 rule](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/naming-rules#example-const-modifier-includes-static-and-readonly)
- XML Documentation on `Logger` class
- Added .editorconfig for unified code styling.

@mahyoussef Looking forward to hearing your feedback.

### CODE REVIEW

File: C#/singleton/Logger.cs [hunk 1]
```diff
@@ -1,10 +1,70 @@
...
+///
+/// The Singleton should always be a 'sealed' class to prevent class
+/// inheritance through external classes and also through nested classes.
+/// </summary>
+public sealed class Logger
 {
-    public enum LogLevel
+    private static Logger? s_instance;
+    private LogLevel _logLevel = LogLevel.INFO;
+    private readonly static object LockObject = new();
```
Comment (mahyoussef) [inline C#/singleton/Logger.cs:15]: I can't remember exactly the convention for writing read-only fields, but shouldn't that be the same as for all private members? 

```C#
private readonly static object _lockObject = new();
```

**Comment by mahyoussef** [comment]
Great job, Youssef! I really loved that you made the Logger class sealed for inheritance & Nested Class Prohibition. Keep up the great work 🚀❤️

**Comment by YoussefWaelMohamedLotfy** [comment]
@mahyoussef Regarding the upcoming changes, how would you like me to approach for the others? I was planning to submit a new PR for each design pattern and be merged on `main`? Should I proceed with this way? If yes, then kindly approve the PR and merge and I am glad to proceed. Thanks.

File: C#/singleton/Logger.cs (hunk 1 above)
Comment (mahyoussef) [reply C#/singleton/Logger.cs:15]: @YoussefWaelMohamedLotfy here it's

**Comment by mahyoussef** [comment]
Yes, of course. Feel free to proceed with one PR for a separate design pattern; I think this is the best way. I just left one comment in this PR, please take a quick look into it 😍

**Comment by YoussefWaelMohamedLotfy** [comment]
@mahyoussef I don't see any comments from you. Where can I find your comment?

File: C#/singleton/Logger.cs (hunk 1 above)
Comment (mahyoussef) [reply C#/singleton/Logger.cs:15]: Did it appear now? @YoussefWaelMohamedLotfy  🏃‍♂️

Comment (YoussefWaelMohamedLotfy) [reply C#/singleton/Logger.cs:15]: @mahyoussef this is from VS IDE1006 Rule mentioned in the link above in the description.
image

image

Reviewed the suggestion and just applied it.

**Pull request merged**